sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Wait.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms;
+
+import static org.apache.beam.sdk.transforms.Contextful.fn;
+import static org.apache.beam.sdk.transforms.Requirements.requiresSideInputs;
+
+import com.google.common.collect.Lists;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.transforms.windowing.Never;
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionView;
+
+/**
+ * Allows sequencing of PTransforms, "apply this transform only after that PCollection is ready".
```

Comment:
<!--new_thread; commit:f7ea3853137d4b6fc908c95dfb443907983c43d4; resolved:0-->
This one-liner seems a bit misleading. What does it mean to be ready? I think the details make it clear, but the summary line should be more technically rigorous. And the phrasing "allows ..." is usually redundant with just an active rephrasing. Perhaps something like 

> Delays processing of each window in a PCollection until signaled.
>
> Given a main PCollection and a signal PCollection, produces output identical to its main input but all elements for a window are produced only once that window is closed in the signal PCollection".